diff --git a/target/linux/ramips/image/mt7621.mk b/target/linux/ramips/image/mt7621.mk
index 811146d..84ed9f5 100644
--- a/target/linux/ramips/image/mt7621.mk
+++ b/target/linux/ramips/image/mt7621.mk
@@ -255,8 +255,9 @@ TARGET_DEVICES += iodata_wn-gx300gr
 
 define Device/k2p
   DTS := K2P
-  IMAGE_SIZE := $(ralink_default_fw_size_16M)
+  IMAGE_SIZE := 32448k
   DEVICE_TITLE := Phicomm K2P
+  DEVICE_PACKAGES := kmod-mt7615d luci-app-mtwifi
 endef
 TARGET_DEVICES += k2p
